node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/RealNodeRepository.java

```diff
@@ -65,35 +60,69 @@ public RealNodeRepository(ConfigServerApi configServerApi) {
                 .collect(Collectors.toList());
     }
 
+
+    @Override
+    public List<NodeSpec> getNodes(NodeType... nodeTypes) {
+        if (nodeTypes.length == 0)
+            throw new IllegalArgumentException("Must specify at least 1 node type");
+
+        return getNodes(Optional.empty(), Arrays.asList(nodeTypes));
+    }
+
     @Override
     public Optional<NodeSpec> getNode(String hostName) {
         try {
             GetNodesResponse.Node nodeResponse = configServerApi.get("/nodes/v2/node/" + hostName,
-                                                                     GetNodesResponse.Node.class);
+                    GetNodesResponse.Node.class);
             if (nodeResponse == null) {
                 return Optional.empty();
             }
             return Optional.of(createNodeRepositoryNode(nodeResponse));
-        } catch (HttpException.NotFoundException|HttpException.ForbiddenException e) {
+        } catch (HttpException.NotFoundException | HttpException.ForbiddenException e) {
             // Return empty on 403 in addition to 404 as it likely means we're trying to access a node that
             // has been deleted. When a node is deleted, the parent-child relationship no longer exists and
             // authorization cannot be granted.
             return Optional.empty();
         }
     }
 
+    /**
+     * Get all ACLs that belongs to a hostname. Usually this is a parent host and all
+     * ACLs for child nodes are returned.
+     */
     @Override
-    public List<AclSpec> getNodesAcl(String hostName) {
+    public Map<String, Acl> getAcls(String hostName) {
+        Map<String, Acl> acls = new HashMap<>();
         try {
             final String path = String.format("/nodes/v2/acl/%s?children=true", hostName);
             final GetAclResponse response = configServerApi.get(path, GetAclResponse.class);
-            return response.trustedNodes.stream()
-                    .map(node -> new AclSpec(
-                            node.hostname, node.ipAddress, ContainerName.fromHostname(node.trustedBy)))
-                    .collect(Collectors.toList());
+
+            // Group ports by container hostname that trusts them
+            Map<String, List<GetAclResponse.Port>> trustedPorts = response.trustedPorts.stream()
+                    .collect(Collectors.groupingBy(GetAclResponse.Port::getTrustedBy));
+
+            // Group nodes by container hostname that trusts them
+            Map<String, List<GetAclResponse.Node>> trustedNodes = response.trustedNodes.stream()
+                    .collect(Collectors.groupingBy(GetAclResponse.Node::getTrustedBy));
+
+            // For each hostname create an ACL
+            Stream.of(trustedNodes.keySet(), trustedPorts.keySet())
+                    .flatMap(Set::stream)
+                    .forEach(hostname -> acls.put(hostname,
```

Comment:
Add `.distinct()` before `.forEach()` since there's likely overlap between the two key sets.